lib/segment/src/vector_storage/query_scorer/mod.rs

```diff
@@ -29,7 +30,8 @@ pub fn score_max_similarity<TMetric: Metric<VectorElementType>>(
         sum += multi_dense_b
             .iter()
             .map(|dense_b| TMetric::similarity(dense_a, dense_b))
-            .fold(ScoreType::NEG_INFINITY, |a, b| if a > b { a } else { b });
+            .max_by(|a, b| OrderedFloat(*a).cmp(&OrderedFloat(*b)))
+            .unwrap_or(ScoreType::NEG_INFINITY);
```

Comment:
We can just `unwrap` it and add `assert` with non-empty check. `NEG_INFINITY` was for `fold` only, not needed to return it if multivector is empty

Reply:
Done with an extra TODO to remind us to add the validation https://github.com/qdrant/qdrant/pull/3973/commits/2397628dbc159e0d7a32a821a97061affcd23930